datasets/senti_lex/senti_lex.py

```diff
@@ -192,27 +190,37 @@ def _split_generators(self, dl_manager):
             datasets.SplitGenerator(
                 name=datasets.Split.TRAIN,
                 gen_kwargs={
-                    "filepaths_dict": [
+                    "filepaths": [
                         [
                             os.path.join(data_dir, "sentiment-lexicons/" + i)
-                            for i in os.listdir(data_dir + "/sentiment-lexicons")
-                            if j in i
+                            for i in os.listdir(os.path.join(data_dir, "sentiment-lexicons"))
+                            if (j + ".txt") == i.split("_")[len(i.split("_")) - 1]
                         ]
                         for j in LANGS
-                    ]
+                    ],
+                    "data_dir": data_dir,
                 },
             ),
         ]
 
-    def _generate_examples(self, filepaths_dict):
+    def _generate_examples(self, filepaths, data_dir):
         """ Yields examples. """
 
-        filepaths_dict = [i for i in filepaths_dict if self.config.name in i]
-        print(filepaths_dict)
-        for filepath in filepaths_dict:
-            print(filepath)
+        filepaths = [
+            i
+            for i in filepaths
+            if [
+                os.path.join(data_dir, "sentiment-lexicons/negative_words_" + self.config.name + ".txt"),
+                os.path.join(data_dir, "sentiment-lexicons/positive_words_" + self.config.name + ".txt"),
+            ]
+            == sorted(i)
+        ]
```

Comment:
Why not simply do that ?
```python
filepaths = [
    os.path.join(data_dir, "sentiment-lexicons", "negative_words_" + self.config.name + ".txt"),
    os.path.join(data_dir, "sentiment-lexicons", "positive_words_" + self.config.name + ".txt"),
]
```

This way you don't have to use listdir or the list of list of paths and you could only pass `data_dir` in `gen_kwargs`

Reply:
Great suggestion